### DISCUSSION TITLE
How do I set allow_unsigned_extensions when using httpfs, postgres and sqlite extensions in the duckdb CLI?

### DISCUSSION THREAD

**Discussion by mskyttner** [duckdb/duckdb#4959] [Q&A]
I'm trying to use the exciting scanners to connect to data stored in postgres and sqlite through the duckdb CLI. 

I build from the main branch and use Docker, but got stuck on:

```bash
#27 [stage-1 11/19] RUN duckdb -unsigned myduck.db "load '/extensions/tpch.duckdb_extension'; call dbgen(sf=0.1);"
#27 DONE 3.2s

#28 [stage-1 12/19] RUN duckdb -unsigned myduck.db 'select 42;'
#28 0.448 ┌────┐
#28 0.448 │ 42 │
#28 0.448 ├────┤
#28 0.448 │ 42 │
#28 0.448 └────┘
#28 DONE 0.5s

#29 [stage-1 13/19] RUN duckdb -unsigned myduck.db 'load "/extensions/httpfs.duckdb_extension";'
#29 0.472 Error: IO Error: Extension "/extensions/httpfs.duckdb_extension" could not be loaded because its signature is either missing or invalid and unsigned extensions are disabled by configuration (allow_unsigned_extensions)
```

So, what seems to work for tpch.duckdb_extension (to load an unsigned extension) does not seem to work for the httpfs.duckdb_extension....

I looked at https://duckdb.org/docs/sql/configuration and I'm not sure how and where to set "allow_unsigned_extensions"?

I tried this but it doesn't work:

`duckdb -unsigned myduck.db 'SET enable_external_access=true; load "/extensions/httpfs.duckdb_extension";':
#30 0.323 Error: Invalid Input Error: Cannot change enable_external_access setting while database is running`

Is it a build time setting or a runtime setting?

And can I load the postgres and sqlite extensions using the duckdb CLI from the build on the main branch?

**Comment by pdet** [comment]
For the CLI you have to pass -unsigned as a command line parameter.

**Reply by mskyttner** [reply]
Thanks for the hint! I try to do that above and it works for tpch, but not for httpfs, despite passing -unsigned?

**Reply by pdet** [reply]
Ah you are already doing that actually, maybe that's due to you issuing multiple cli runs?

Can you try 
``` 
RUN duckdb -unsigned myduck.db "load '/extensions/tpch.duckdb_extension'; call dbgen(sf=0.1); select 42; load '/extensions/httpfs.duckdb_extension';"
```

**Reply by mskyttner** [reply]
This works, thanks! It gets me one step further, but I'm still stuck.... hmmm... I wonder why several runs seems to affect the behavior here? I would like to be able to do multiple CLI executions, so how can I work around this...? 

It looks like the first time it works OK. Could it be because the myduckd.db file does not yet exist at that point, but the second time (or subsequent times) it fails because now the database file exists?

**Reply by Mytherin** [reply]
Are you running on the latest master? This looks like it might be related to https://github.com/duckdb/duckdb/pull/4895

**Reply by mskyttner** [reply]
It looks related, not sure, and yes, I'm running on master from earlier today. I can replicate it using this:

```bash
root@fd0cda5b3216:/data# duckdb -unsigned test.db "load '/extensions/httpfs.duckdb_extension';"
root@fd0cda5b3216:/data# duckdb -unsigned test.db "load '/extensions/httpfs.duckdb_extension';"
Error: IO Error: Extension "/extensions/httpfs.duckdb_extension" could not be loaded because its signature is either missing or invalid and unsigned extensions are disabled by configuration (allow_unsigned_extensions)
```
First run is OK, second not. Feels like it has to do with whether the test.db file exists or not?
